src/scala/com/github/johnynek/bazel_deps/Target.scala

```diff
@@ -76,25 +77,52 @@ case class Target(
 
     val targetType = Doc.text(s"${langName}_${kind}")
 
-    def sortKeys(items: List[(String, Doc)]): Doc = {
+    def sortKeys(tt: Doc, name: String, items: List[(String, Doc)]): Doc = {
       // everything has a name
-      val nm = ("name", quote(name.name))
+      val nm = ("name", quote(name))
       implicit val ordDoc: Ordering[Doc] = Ordering.by { d: Doc => d.renderWideStream.mkString }
       val sorted = items.collect { case (s, d) if !(d.isEmpty) => (s, d) }.sorted
 
-      renderList(targetType + Doc.text("("), nm :: sorted, Doc.text(")")) { case (k, v) =>
+      renderList(tt + Doc.text("("), nm :: sorted, Doc.text(")")) { case (k, v) =>
         k +: " = " +: v
-      } + Doc.line.repeat(2)
+      } + Doc.line
     }
 
     def labelList(ls: Set[Label]): Doc =
       renderList(Doc.text("["), ls.toList.map(_.asStringFrom(name.path)).sorted, Doc.text("]"))(quote)
 
-    sortKeys(List(
+    def renderExportedPlugins(pcs: Set[ProcessorClass]): Doc = {
+      renderList(Doc.text("["), pcs.toList.map(pc => ":" + getPluginTargetName(pcs, pc)).sorted, Doc.text("]"))(quote)
+    }
+    def getPluginTargetName(pcs: Set[ProcessorClass], processorClass: ProcessorClass) =
+      if (pcs.size == 1) s"${name.name}_plugin"
+      else {
+        val fqnToSnakeCase = camel2Underscore(shortName(processorClass.asString))
+        s"${name.name}_plugin_$fqnToSnakeCase"
+      }
+    def camel2Underscore(text: String) = text.drop(1).foldLeft(text.headOption.map(_.toLower + "") getOrElse "") {
+      case (acc, c) if c.isUpper => acc + "_" + c.toLower
+      case (acc, c) if c == '$' => acc
+      case (acc, c) => acc + c
+    }
+    def shortName(fqn: String) = if (fqn.contains(".")) fqn.substring(fqn.lastIndexOf('.') + 1) else fqn
```

Comment:
Will the last component always be unique?

Reply:
Hmm. Potentially you could have `v1.Processor` and `v2.Processor` in the same `java_library`. It seems better then to use the fully-qualified name if there's more than one plugin class to make it simple. In that case, transforming camelCase class names to snake case probably doesn't make sense.